### DISCUSSION TITLE
Add an option to manage hierarchy for stories

### DISCUSSION THREAD

**Discussion by ghost** [storybookjs/storybook#24307] [Ideas]
### Is your feature request related to a problem? Please describe.

I want to have a structure like this one 

| Component
- Docs
- Playground (one story with all the variants)
+ Stories (a list of stories per variant)

![image](https://github.com/storybookjs/storybook/assets/128583596/cfa77e39-97a8-4161-8bda-c3e18e2981e6)
![image](https://github.com/storybookjs/storybook/assets/128583596/4af06aa9-c826-48da-951f-bccbc3c4cde7)

_________

In order to achieve this I need to create three files for every component: one for Docs, one for the Playground story, and one for the list of Stories.

This is very annoying since implies a lot of mindless work and a lot of code repetition since we also want to rely on autodocs.

### Describe the solution you'd like

I would like to have a way to manage the folder structure for the stories in order to be able to achieve this structure using only one file. 

I don't mind if the logic for this is not located in the .stories.tsx file, it could be in manager.ts or preview.ts, the same way is implemented the renderLabel function in config.

### Describe alternatives you've considered

I've tried many workarounds for this, decorators and addons but I have not found a method to mutate the title of the stories inside a story. I'm not sure if the indexer function could be applied to solve this.

### Are you able to assist to bring the feature to reality?

no

**Comment by enix79** [comment]
+1 

It would be nice if we could add a "Title"/Path per story instead of a per component basis. Sometimes you have dozens of stories and you want to structure them in a bit more convenient way. If coupling stories to components is required, one could work with relative paths.
